FtpClient/FtpCli/Pkgs/SftpClient.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using Renci.SshNet;
+using Renci.SshNet.Sftp;
+
+namespace FtpCli.Packages.ClientWrapper
+{
+
+    public class Client : IClient
+    {
+        private SftpClient _client;
+        private ConnectionInfo _connection;
+        private bool _isConnected;
+        private delegate void UseCallback(SftpClient client);
+
+        // Explicit constructor
+        // Should be used in all cases
+        public Client(string host, int port, string username, string password)
+        {
+             _connection = new ConnectionInfo(host, port, username,
+                 new PasswordAuthenticationMethod(username, password)); 
+
+             // All operations will be performed through the '_client'
+             _client = new SftpClient(_connection);
+
+             try {
+                _client.Connect();
+                _isConnected = true;
+             } catch (Exception err) 
+             {
+                Console.WriteLine($"Unable to connect to {host}:{port}");
+                Console.WriteLine(err.ToString());
+                Environment.Exit(-1);
```

Comment:
Do we want to exit here? It might be better to handle the program execution logic as near to the main method as possible so we don't clutter the execution with arbitrary exit codes around the program? Could we just throw this error?

Reply:
We could. My main reason for putting this there was I didn't see the need for further execution if the client is unable to connect.

If this was to be made throwing an exception, we (the team in general) should figure out how deep we want to nest try/catch blocks. I think having them been in the Logic Layer would make the most sense, but again we can all discuss it.